google-cloud-storage/src/main/java/com/google/cloud/storage/StorageImpl.java

```diff
@@ -476,7 +531,10 @@ public CopyWriter copy(final CopyRequest copyRequest) {
   public byte[] readAllBytes(BlobId blob, BlobSourceOption... options) {
     final StorageObject storageObject = blob.toPb();
     final Map<StorageRpc.Option, ?> optionsMap = optionMap(blob, options);
-    return run(() -> storageRpc.load(storageObject, optionsMap), Function.identity());
+    ExceptionHandler exceptionHandler =
+        retryAlgorithmManager.getForObjectsGet(storageObject, optionsMap);
+    return run(
```

Comment:
Question on design, I left a comment on the design document. I'm wondering if we can reduce complexity here a little bit for future us.

Reply:
Update: @BenWhitehead and I spoke on Friday through the current design and decided to move forward with the existing design for the following reasons:

1. It's not explicitly exposed as a Public API (yet).
2. It refactors the complexity of Resumable Media operations to map back to gRPC API spec which helps clean up internal maintanence
3. We would like to make progress towards Retry Strategy alignment while acknowledging there's not enough information on gotchas from introducing gRPC Retries yet. We are making decisions with the information we have right now.